mesonbuild/build.py

```diff
@@ -669,6 +670,10 @@ def process_kwargs(self, kwargs, environment):
         for i in self.link_args:
             if not isinstance(i, str):
                 raise InvalidArguments('Link_args arguments must be strings.')
+        for l in self.link_args:
+            if '-Wl,-rpath' in l:
```

Comment:
Converted slightly stricter. We can always make it more strict in the future but this should get 99% of all cases.